### PR TITLE
[node-cron] Define timezone as a string literal

### PR DESCRIPTION
**Pull request by RMHonor**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by burtek** [comment]
I'm not sure that relying on another package's data is a good idea here. Wouldn't it be safer to create typings for `tz-offset` and depend on them here? At least when `tz-offset` data changes, one can change it in `@types/tz-offset` and it would be handled here too...

**Comment by RMHonor** [comment]
Whilst I agree, `tz-offset` is not a package published on NPM. Is it normal to have type libraries published available for libraries which aren't available through standard means?

Happy to make a start on `tz-offset` anyway

**Comment by burtek** [comment]
> Whilst I agree, `tz-offset` is not a package published on NPM

https://www.npmjs.com/package/tz-offset 🤔

**Comment by RMHonor** [comment]
Blind as a bat 😆

**Comment by burtek** [comment]
Happens to me too, np :D

**Comment by RMHonor** [comment]
#34209 may be of interest to you!

**Comment by burtek** [comment]
@RMHonor approved that one :)

**Review by burtek — CHANGES_REQUESTED** [review]
Make it depend on `@types/tz-offset` as per previous discussion

**Comment by burtek** [comment]
@RMHonor any plans for making this happen? :)

**Comment by RMHonor** [comment]
PR updated to use `tz-offset`

**Review by burtek — CHANGES_REQUESTED** [review]
Shouldn't `@types/tz-offset` be added to dependencies in `package.json` now? [It won't be included by default](https://github.com/DefinitelyTyped/DefinitelyTyped#i-notice-some-packages-having-a-packagejson-here).

```json
{
    "private": true,
    "dependencies": {
        "@types/tz-offset": "0.0.0"
    }
}

**Comment by RMHonor** [comment]
Ah, of course. Pushed a new `package.json`

**Review by burtek — APPROVED** [review]

File: types/node-cron/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "private": true,
+  "dependencies": {
+      "tz-offset": "0.0.0"
```
Comment (burtek) [inline types/node-cron/package.json]: `@types/tz-offset`

**Review by burtek — CHANGES_REQUESTED** [review]

File: types/node-cron/package.json (hunk 1 above)
Comment (RMHonor) [reply types/node-cron/package.json]: Updated

**Comment by burtek** [comment]
Actually my bad 🤦‍♂️ it must have changed some time ago
image

**Comment by RMHonor** [comment]
Ah, that's clever, so DefinitelyTyped is able to look at the required imports in `.d.ts` files and adds them as dependencies?

Updated to remove the `package.json`

**Comment by burtek** [comment]
> DefinitelyTyped is able to look at the required imports in `.d.ts` files and adds them as dependencies?

looks like this.

**Review by burtek — APPROVED** [review]

**Comment by burtek** [comment]
@RMHonor it does work ;)
https://www.npmjs.com/package/@types/node-cron?activeTab=dependencies

**Pull request merged**